packages/sdk/src/credential/teamsUserCredential.browser.ts

```diff
@@ -116,12 +116,23 @@ export class TeamsUserCredential implements TokenCredential {
               return;
             }
 
+            let resultJson: any = {};
             try {
-              const accessToken = parseAccessTokenFromAuthCodeTokenResponse(result);
-              resolve(accessToken);
-            } catch (error: any) {
-              reject(error);
+              resultJson = JSON.parse(result);
+            } catch (error) {
+              // If can not parse result as Json, will NOT throw error since user may return other info in auth-end page.
+              // TODO: resolve the result.
+              const failedToParseResult = "Failed to parse result to Json.";
+              internalLogger.verbose(failedToParseResult);
+              resolve();
+            }
+
+            // If sessionStorage exists in result, set the values in current session storage.
+            if (resultJson.sessionStorage) {
```

Comment:
if resultJson.sessionStorage is empty, do we need to throw error or show error messages?